src/geom/Functor.cc

```diff
@@ -0,0 +1,56 @@
+// -*- lsst-c++ -*-
+
+/* 
+ * LSST Data Management System
+ * Copyright 2015 LSST Corporation.
+ * 
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the LSST License Statement and 
+ * the GNU General Public License along with this program.  If not, 
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <cmath>
+#include <iostream>
+
+#include "lsst/pex/exceptions.h"
+#include "lsst/afw/geom/Functor.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+
+Functor::Functor(std::string const & owner) 
+   : daf::base::Citizen(typeid(this)), _owner(owner) {
+}
+
+double Functor::inverse(double y, double tol, unsigned int maxiter) const {
+   /// Use Newton-Raphson method to find the inverse.
```

Comment:
Should you clip x to the valid range for operator()? I worry that this may unexpected fail otherwise.

Also I suggest you sanity-check tol>0 and maxiter>0 before starting the iteration; that may catch errors of mis-ordering arguments.